torch/_subclasses/meta_utils.py

```diff
@@ -380,19 +426,102 @@ def is_c_of_r(complex_dtype, real_dtype):
                         # recreate this situation.
                         def _view_from_base(base, t):
                             if t.is_nested:
-                                # Nested tensors do not support as_strided, and
-                                # hence,always have _view_func available.
-                                #
-                                # The unsafe version of _view_func omits
-                                # checking whether the base passed in has the same
-                                # metadata as the original base the view_func
-                                # was originally executed with. (1) It is OK here,
-                                # because we're calling it on the meta-ified base,
-                                # so the metadata is guaranteed to be the same.
-                                # (2) It is necessary because we don't actually
-                                # want to guard on the base's metadata here.
-                                return t._view_func_unsafe(base)
+                                if base.is_nested:
+                                    # Nested tensors do not support as_strided, and
+                                    # hence,always have _view_func available.
+                                    #
+                                    # The unsafe version of _view_func omits
+                                    # checking whether the base passed in has the same
+                                    # metadata as the original base the view_func
+                                    # was originally executed with. (1) It is OK here,
+                                    # because we're calling it on the meta-ified base,
+                                    # so the metadata is guaranteed to be the same.
+                                    # (2) It is necessary because we don't actually
+                                    # want to guard on the base's metadata here.
+                                    return t._view_func_unsafe(base)
+                                else:
+                                    from torch._dynamo.source import AttrSource
+                                    from torch.nested._internal.nested_tensor import (
+                                        _tensor_symint_registry,
+                                        nested_view_from_values_offsets,
+                                        nested_view_from_values_offsets_lengths,
+                                    )
+
+                                    # Meta-ify values while maintaining proper view
+                                    # relationship with the base
+                                    view_nt = t._view_func_unsafe(base)
```

Comment:
Maybe add a short note that we need to replay this view_func so that chained view_func are properly replayed.

Reply:
Shouldn't be relevant anymore